File: bot.js
const mineflayer = require('mineflayer')
const express = require('express')

// Веб-сервер чтобы Render не спал
const app = express()
app.get('/', (req, res) => res.send('Bot is alive!'))
app.listen(8080, () => console.log('Web server running'))

// Настройки — замени HOST на адрес твоего Aternos сервера
const HOST = 'megasosalka.aternos.me'
const PORT = 16429
const USERNAME = 'AFK_Bot'

function createBot() {
  const bot = mineflayer.createBot({
    host: HOST,
    port: PORT,
    username: USERNAME,
    version: '1.20.1'
  })

  bot.on('spawn', () => console.log('Бот зашёл на сервер!'))
  bot.on('kicked', (reason) => {
    console.log('Кикнули:', reason)
    setTimeout(createBot, 5000) // переподключение через 5 сек
  })
  bot.on('error', (err) => {
    console.log('Ошибка:', err)
    setTimeout(createBot, 5000)
  })
  bot.on('end', () => {
    console.log('Отключился, переподключаюсь...')
    setTimeout(createBot, 5000)
  })
}

createBot()

File: bot.py
import minecraft_launcher_lib
import mineflayer
from flask import Flask
from threading import Thread

# Веб-сервер чтобы Render не усыплял
app = Flask('')

@app.route('/')
def home():
    return "Bot is alive!"

def run_web():
    app.run(host='0.0.0.0', port=8080)

Thread(target=run_web).start()

# Настройки — замени на свои
HOST = "megasosalka.aternos.me"
PORT = 16429
USERNAME = "AFK_Bot"

bot = mineflayer.createBot({
    'host': HOST,
    'port': PORT,
    'username': USERNAME,
    'version': '1.21.11'  # версия твоего сервера
})

@bot.on('spawn')
def on_spawn():
    print("Бот зашёл на сервер!")

@bot.on('kicked')
def on_kicked(reason):
    print(f"Бота кикнули: {reason}")

@bot.on('error')
def on_error(err):
    print(f"Ошибка: {err}")
